bulk-ingester/src/test/java/org/alfresco/hxi_connector/bulk_ingester/processor/mapper/AlfrescoPropertyMapperTest.java

```diff
@@ -0,0 +1,489 @@
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+
+package org.alfresco.hxi_connector.bulk_ingester.processor.mapper;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import static org.alfresco.elasticsearch.db.connector.model.PropertyValueType.BOOLEAN;
+import static org.alfresco.elasticsearch.db.connector.model.PropertyValueType.CONTENT_DATA_ID;
+import static org.alfresco.elasticsearch.db.connector.model.PropertyValueType.DATE;
+import static org.alfresco.elasticsearch.db.connector.model.PropertyValueType.DOUBLE;
+import static org.alfresco.elasticsearch.db.connector.model.PropertyValueType.FLOAT;
+import static org.alfresco.elasticsearch.db.connector.model.PropertyValueType.LONG;
+import static org.alfresco.elasticsearch.db.connector.model.PropertyValueType.NODEREF;
+import static org.alfresco.elasticsearch.db.connector.model.PropertyValueType.NULL;
+import static org.alfresco.elasticsearch.db.connector.model.PropertyValueType.PATH;
+import static org.alfresco.elasticsearch.db.connector.model.PropertyValueType.SERIALIZABLE;
+import static org.alfresco.elasticsearch.db.connector.model.PropertyValueType.STRING;
+
+import java.time.ZonedDateTime;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.util.SerializationUtils;
+
+import org.alfresco.elasticsearch.db.connector.model.AlfrescoNode;
+import org.alfresco.elasticsearch.db.connector.model.ContentMetadata;
+import org.alfresco.elasticsearch.db.connector.model.NodeProperty;
+import org.alfresco.elasticsearch.db.connector.model.PropertyKey;
+import org.alfresco.elasticsearch.db.connector.model.PropertyValue;
+import org.alfresco.elasticsearch.db.connector.model.PropertyValueType;
+
+@SuppressWarnings({"PMD.TooManyMethods", "PMD.LinguisticNaming"})
+class AlfrescoPropertyMapperTest
+{
+    @Test
+    void shouldMapJustPropertiesWithGivenName()
+    {
+        // given
+        AlfrescoNode alfrescoNode = new AlfrescoNode();
+
+        String description = "description";
+        String descriptionText = "The purpose of document is...";
+
+        alfrescoNode.setNodeProperties(
+                Set.of(
+                        createNodeProperty("title"),
+                        createNodeProperty(description, stringValue(descriptionText))));
+
+        // when
+        var property = new AlfrescoPropertyMapper(alfrescoNode, description).performMapping();
+
+        // then
+        assertTrue(property.isPresent());
+        assertEquals(description, property.get().getKey());
+        assertEquals(descriptionText, property.get().getValue());
```

Comment:
Can we replace the three statements with a single equality check like:
```
assertEquals(expectedOptional, property);
```

Reply:
good point! fixed